hydroflow/src/builder/build/mod.rs

```diff
@@ -10,6 +10,7 @@ pub mod pull_handoff;
 pub mod pull_iter;
 pub mod pull_join;
 pub mod pull_map;
+pub mod pull_stream_join;
```

Comment:
StreamJoin is an ambiguous name. This thing is asymmetric: one side is fully streaming/transient, the other side accumulates over time. I'd call it something like OneSidedHashJoin (as opposed to SymmetricHashJoin) or  StreamBufferHashJoin (in the old streaming database papers I think it'd be called a Stream-Table Join).